### PR TITLE
[154] Rewrite tests to use pynuoadmin rather than entity

### PR DESCRIPTION
**Pull request by madscientist**
In NuoDB 4.0 and above the previous NuoDB Agent administrative layer is
deprecated in favor of the NuoDB Admin layer, and in NuoDB 4.2 and above
NuoDB Agent is obsolete.

This means that the pynuodb "entity" layer is no longer useful: instead
users should use the pynuoadmin module to manage NuoDB databases.

Further, NuoDB 4.2 provides a new Vectorized Execution Engine as its
preferred SQL interface.

This commit modifies the tests for this new world.

First, it removes the entity tests altogether as it's not possible (as
of NuoDB 4.2) to use them with a database.

Second, it creates module-level operations to re-implement the
entity-based database management (create/start/stop) used by the tests
using the pynuoadmin interface.

Unfortunately pynuoadmin still requires Python 2, while pynuodb supports
both Python 2 and 3, so also re-implement the database management using
the nuocmd CLI so that the tests will run with Python 3.  This is the
implementation which is currently used.

Finally, VEE removes some support for obsolete features (such as the
"number" SQL type) which were used by the tests; remove references to
them.

### CODE REVIEW

**Comment by madscientist** [comment]
This is the right one.

File: .travis.yml [hunk 1]
```diff
@@ -21,20 +21,19 @@ notifications:
...
+  - wget -q "$NUO_DOWNLOAD/${NUO_PREFIX}_$(tail -n1 /tmp/supportedversions.txt)_amd64.deb" --output-document=/var/tmp/nuodb.deb
   - sudo dpkg -i /var/tmp/nuodb.deb
 
 install:
   - make install
 
 before_script:
-  - if [ "$NUO_ADD_DOMAIN_PASS" = yes ]; then printf '\ndomainPassword = bird\n' | sudo tee -a "$NUODB_HOME"/etc/default.properties >/dev/null; fi
-  - if [ "$NUO_START_AGENT" = yes ]; then sudo service nuoagent start; fi
+  - make start-nuoadmin NUO_ENABLE_TLS=$NUO_ENABLE_TLS
```
Comment (madscientist) [inline .travis.yml:32]: After 4.2 is released we can use nuoadmin tls disable etc. here but the sed script we need in 4.1 is too hard to implement directly in a yaml statement due to the backslashes etc. so put it into the makefile.

File: tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+db_created = False
+db_started = False
+
+
+def waitforsql(tmout):
+    import pynuodb
+    host = 'localhost'
+    port = os.environ.get('NUODB_PORT')
+    if port:
+        host += ':' + port
```
Comment (madscientist) [inline tests/__init__.py:39]: This method of determining the agent address is not great: ideally we'd use the same server we use to start/stop things and retrieve the proper address and port from that.  This info is available but it's not easy to get.

**Comment by mkysel** [comment]
I am assuming that you want to merge to master? Not your other private branch? 
```
madscientist wants to merge 1 commit into pds/155 from pds/154
```

**Comment by mkysel** [comment]
I am assuming that you want this to merge to master?
```
madscientist wants to merge 1 commit into pds/155 from pds/154
```

**Comment by madscientist** [comment]
> I am assuming that you want to merge to master? Not your other private branch?

Yes, it will all work out in the end :).  I needed to base this change on the other change, which hasn't yet been pushed, in order to get it working.  I'll push them both together or, at least, the other one first.

**Comment by mkysel** [comment]
we have not released a version of pynuodb here on GH since 2015. Are we planning on releasing a new one?

Do the changes here only work with 4.2 or newer?

**Comment by madscientist** [comment]
We haven't made a release, but we've added tags that you can use instead (click the "Tags" to switch from releases to tags).  And we put the tagged versions on pypi.  The latest such was 2.5.2.

I don't know why we didn't make releases for these tags; they are intended to be releases.

Yes, this works with pre-4.2; it works with anything 4.0 and above (requires nuoadmin).  In fact, you can be sure of that because when I ran it in Travis over the weekend 4.2 was not released yet so it ran against 4.1.2 :).

**Review by mkysel — APPROVED** [review]
good to see old admin gone!

**Comment by mkysel** [comment]
you should also change the README. It mentions the entity

**Comment by madscientist** [comment]
entity is still there, it's just we can't use it in the tests anymore so it's not tested.  If people want to use pynuodb with pre-4.2 releases and still use NuoDB Agent they can do so (in fact we still have internal tests that do this so we can't delete it yet) and it should work as well as it did as before.

But I will definitely adjust the README, that's a good idea I didn't look at it.

**Pull request closed without merging**